lib/puppet/provider/package/yum.rb

```diff
@@ -25,6 +25,14 @@
 
   defaultfor :osfamily => :redhat
 
+  def execute(*args)
```

Comment:
Yeah, maybe this would be useful in the type so we can use it in multiple providers. Or in a common base-class for the providers.